You are a medical analyst working in a heart center hospital. You get patient checkup data from the administration. As a health analyst, you are asked to take insight from a given dataset. Before analyzing, you must process the data first.

  Using dataset (heart.csv) from https://www.kaggle.com/rashikrahmanpritom/heart-attack-analysis-prediction-dataset?select=heart.csv, Obtain patient data with the following criteria:
  - Age **above average** data
  - Sex **Male** (Sex=1)
  - Cholesterol below 150 mg/dl
  - Maximum heart rate achieved is not more than 180


  **Note**: *Description of column names can be seen in the link provided*

In [2]:
# Import libraries needed
import numpy as np
import pandas as pd

In [3]:
# Data Loading
# A simple way to store big data sets is to use CSV files
df = pd.read_csv('heart.csv')

In [4]:
# simple data exploratory
# used to get the first n rows. This function returns the first n rows for the object based on position
df.head()

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0   63    1   3     145   233    1        0       150     0      2.3    0   
1   37    1   2     130   250    0        1       187     0      3.5    0   
2   41    0   1     130   204    0        0       172     0      1.4    2   
3   56    1   1     120   236    0        1       178     0      0.8    2   
4   57    0   0     120   354    0        1       163     1      0.6    2   

   caa  thall  output  
0    0      1       1  
1    0      2       1  
2    0      2       1  
3    0      2       1  
4    0      2       1

In [5]:
df.tail()
# used to get the last n rows. This function returns last n rows from the object based on position.

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
298   57    0   0     140   241    0        1       123     1      0.2    1   
299   45    1   3     110   264    0        1       132     0      1.2    1   
300   68    1   0     144   193    1        1       141     0      3.4    1   
301   57    1   0     130   131    0        1       115     1      1.2    1   
302   57    0   1     130   236    0        0       174     0      0.0    1   

     caa  thall  output  
298    0      3       0  
299    0      3       0  
300    2      3       0  
301    1      3       0  
302    1      2       0

In [6]:
df.info()
# used to print a concise summary of a DataFrame. This method prints information about a DataFrame including the index 
# dtype and column dtypes, non-null values and memory usage.

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       303 non-null    int64  
 1   sex       303 non-null    int64  
 2   cp        303 non-null    int64  
 3   trtbps    303 non-null    int64  
 4   chol      303 non-null    int64  
 5   fbs       303 non-null    int64  
 6   restecg   303 non-null    int64  
 7   thalachh  303 non-null    int64  
 8   exng      303 non-null    int64  
 9   oldpeak   303 non-null    float64
 10  slp       303 non-null    int64  
 11  caa       303 non-null    int64  
 12  thall     303 non-null    int64  
 13  output    303 non-null    int64  
dtypes: float64(1), int64(13)
memory usage: 33.3 KB


In [7]:
df.describe()
# used to view some basic statistical details like percentile, mean, std etc. of a data frame or a series of numeric 
# values.

age         sex          cp      trtbps        chol         fbs  \
count  303.000000  303.000000  303.000000  303.000000  303.000000  303.000000   
mean    54.366337    0.683168    0.966997  131.623762  246.264026    0.148515   
std      9.082101    0.466011    1.032052   17.538143   51.830751    0.356198   
min     29.000000    0.000000    0.000000   94.000000  126.000000    0.000000   
25%     47.500000    0.000000    0.000000  120.000000  211.000000    0.000000   
50%     55.000000    1.000000    1.000000  130.000000  240.000000    0.000000   
75%     61.000000    1.000000    2.000000  140.000000  274.500000    0.000000   
max     77.000000    1.000000    3.000000  200.000000  564.000000    1.000000   

          restecg    thalachh        exng     oldpeak         slp         caa  \
count  303.000000  303.000000  303.000000  303.000000  303.000000  303.000000   
mean     0.528053  149.646865    0.326733    1.039604    1.399340    0.729373   
std      0.525860   22.905161    0.469794    1.161075    0.616226    1.022606   
min      0.000000   71.000000    0.000000    0.000000    0.000000    0.000000   
25%      0.000000  133.500000    0.000000    0.000000    1.000000    0.000000   
50%      1.000000  153.000000    0.000000    0.800000    1.000000    0.000000   
75%      1.000000  166.000000    1.000000    1.600000    2.000000    1.000000   
max      2.000000  202.000000    1.000000    6.200000    2.000000    4.000000   

            thall      output  
count  303.000000  303.000000  
mean     2.313531    0.544554  
std      0.612277    0.498835  
min      0.000000    0.000000  
25%      2.000000    0.000000  
50%      2.000000    1.000000  
75%      3.000000    1.000000  
max      3.000000    1.000000

#### Above average ages

In [10]:
# Age average
df['age'].mean()

54.366336633663366

In [8]:
# To get the above 54 age, we could run this method :
ageAverage = df['age'] > 54.366336633663366
df[ageAverage]

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0     63    1   3     145   233    1        0       150     0      2.3    0   
3     56    1   1     120   236    0        1       178     0      0.8    2   
4     57    0   0     120   354    0        1       163     1      0.6    2   
5     57    1   0     140   192    0        1       148     0      0.4    1   
6     56    0   1     140   294    0        0       153     0      1.3    1   
..   ...  ...  ..     ...   ...  ...      ...       ...   ...      ...  ...   
297   59    1   0     164   176    1        0        90     0      1.0    1   
298   57    0   0     140   241    0        1       123     1      0.2    1   
300   68    1   0     144   193    1        1       141     0      3.4    1   
301   57    1   0     130   131    0        1       115     1      1.2    1   
302   57    0   1     130   236    0        0       174     0      0.0    1   

     caa  thall  output  
0      0      1       1  
3      0      2       1  
4      0      2       1  
5      0      1       1  
6      0      2       1  
..   ...    ...     ...  
297    2      1       0  
298    0      3       0  
300    2      3       0  
301    1      3       0  
302    1      2       0  

[159 rows x 14 columns]

there are **159** patiets above the age of 54

#### Male (Sex=1)

In [9]:
# Get to know the male patiens :

malePatiens = df.sex ==1
df[malePatiens]

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0     63    1   3     145   233    1        0       150     0      2.3    0   
1     37    1   2     130   250    0        1       187     0      3.5    0   
3     56    1   1     120   236    0        1       178     0      0.8    2   
5     57    1   0     140   192    0        1       148     0      0.4    1   
7     44    1   1     120   263    0        1       173     0      0.0    2   
..   ...  ...  ..     ...   ...  ...      ...       ...   ...      ...  ...   
295   63    1   0     140   187    0        0       144     1      4.0    2   
297   59    1   0     164   176    1        0        90     0      1.0    1   
299   45    1   3     110   264    0        1       132     0      1.2    1   
300   68    1   0     144   193    1        1       141     0      3.4    1   
301   57    1   0     130   131    0        1       115     1      1.2    1   

     caa  thall  output  
0      0      1       1  
1      0      2       1  
3      0      2       1  
5      0      1       1  
7      0      3       1  
..   ...    ...     ...  
295    2      3       0  
297    2      1       0  
299    0      3       0  
300    2      3       0  
301    1      3       0  

[207 rows x 14 columns]

#### Cholesterol below 150 mg/dl

In [10]:
# Get to know list of patients with cholesterol below 150 :

cholesterol = df.chol < 150
df[cholesterol]

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
53    44    0   2     108   141    0        1       175     0      0.6    1   
111   57    1   2     150   126    1        1       173     0      0.2    2   
151   71    0   0     112   149    0        1       125     0      1.6    1   
267   49    1   2     118   149    0        0       126     0      0.8    2   
301   57    1   0     130   131    0        1       115     1      1.2    1   

     caa  thall  output  
53     0      2       1  
111    1      3       1  
151    0      2       1  
267    3      2       0  
301    1      3       0

#### Maximum heart rate achieved below 150

In [11]:
# Get to know list of patients with heart rate below 150 : :

heartRate = df.thalachh < 150
df[heartRate]

age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
5     57    1   0     140   192    0        1       148     0      0.4    1   
11    48    0   2     130   275    0        1       139     0      0.2    2   
13    64    1   3     110   211    0        0       144     1      1.8    1   
17    66    0   3     150   226    0        1       114     0      2.6    0   
23    61    1   2     150   243    1        1       137     1      1.0    1   
..   ...  ...  ..     ...   ...  ...      ...       ...   ...      ...  ...   
297   59    1   0     164   176    1        0        90     0      1.0    1   
298   57    0   0     140   241    0        1       123     1      0.2    1   
299   45    1   3     110   264    0        1       132     0      1.2    1   
300   68    1   0     144   193    1        1       141     0      3.4    1   
301   57    1   0     130   131    0        1       115     1      1.2    1   

     caa  thall  output  
5      0      1       1  
11     0      2       1  
13     0      2       1  
17     0      2       1  
23     0      2       1  
..   ...    ...     ...  
297    2      1       0  
298    0      3       0  
299    0      3       0  
300    2      3       0  
301    1      3       0  

[132 rows x 14 columns]